### PR TITLE
Do not require multiple Fulcio certs in the TUF root

### PR DESCRIPTION
**Pull request by rgerganov**
cosign requires both fulcio.crt.pem and fulcio_v1.crt.pem in the TUF
root which doesn't make sense when using local TUF. fulcio_v1.crt.pem
was added in the embedded TUF in order to support Fulcio v1 but it
shouldn't be required when users initialize cosign with their own TUF
repo.

Closes: #1229

Signed-off-by: Radoslav Gerganov <rgerganov@vmware.com>

### CODE REVIEW

**Review by hectorj2f — COMMENTED** [review]
Lgtm

File: cmd/cosign/cli/fulcio/fulcioroots/fulcioroots.go [hunk 1]
```diff
@@ -64,11 +64,16 @@ func initRoots() *x509.CertPool {
 		// Retrieve from the embedded or cached TUF root. If expired, a network
 		// call is made to update the root.
 		ctx := context.Background() // TODO: pass in context?
-		for _, fulcioTarget := range []string{fulcioTargetStr, fulcioV1TargetStr} {
-			buf := tuf.ByteDestination{Buffer: &bytes.Buffer{}}
-			if err := tuf.GetTarget(ctx, fulcioTarget, &buf); err != nil {
-				panic(errors.Wrap(err, "creating root cert pool"))
-			}
+		buf := tuf.ByteDestination{Buffer: &bytes.Buffer{}}
```
Comment (asraa) [inline cmd/cosign/cli/fulcio/fulcioroots/fulcioroots.go:67]: WDYT about keeping the previous code and use a bool to panic if one of the targets wasn't added?
(as in gracefully fail, just fail when none of the errors were nil)

**Review by asraa — COMMENTED** [review]
Thank you so so much for these changes!

I think long term we might want some kind of general discovery like look for targets named `fulcio.*`, and as long as one of them is successfully added, don't panic.

File: cmd/cosign/cli/fulcio/fulcioroots/fulcioroots.go (hunk 1 above)
Comment (rgerganov) [reply cmd/cosign/cli/fulcio/fulcioroots/fulcioroots.go:67]: make sense, I will update the patch

**Review by dlorenc — APPROVED** [review]

**Pull request merged**